syncopy/datatype/base_data.py

```diff
@@ -1931,22 +1952,31 @@ def _selection_setter(self, data, select, dataprop, selectkey):
                 else:
                     targetArr = np.arange(target.size)
                 if not set(selection).issubset(targetArr):
-                    lgl = "list/array of {} existing names or indices".format(dataprop)
+                    lgl = "list/array of {} existing names or indices".format(selectkey)
                     raise SPYValueError(legal=lgl, varname=vname)
 
                 # Preserve order and duplicates of selection - don't use `np.isin` here!
                 idxList = []
                 for sel in selection:
                     idxList += list(np.where(targetArr == sel)[0])
 
-                if dataprop in ["unit", "eventid"]:
-                    setattr(self, selector, getattr(data, "_get_" + dataprop)(self.trials, idxList))
+                if selectkey in ["unit", "eventid"]:
+                    setattr(self, selector, getattr(data, "_get_" + selectkey)(self.trials, idxList))
                 else:
                     # if possible, convert range-arrays (`[0, 1, 2, 3]`) to slices for better performance
                     if len(idxList) > 1:
                         steps = np.diff(idxList)
                         if steps.min() == steps.max() == 1:
                             idxList = slice(idxList[0], idxList[-1] + 1, 1)
+
+                    # be careful w/pairwise list-channel selections in `CrossSpectralData` objects
+                    # (that could not be converted to slices above)
+                    if isinstance(idxList, list) and selectkey in ["channel_i", "channel_j"]:
+                        if len(idxList) > 1:
+                            err = "Multi-channel-pair selections not supported"
+                            raise NotImplementedError(err)
```

Comment:
I think this now is a very clear interface, which at the same time reduces the chance of ppl pumping too much stuff into memory :+1:  ..anything more complicated ppl can/should do themselves by e.g. looping over `.show` statements OR preferably by specifying the interesting channels before doing the cross-measure. I'll try to encourage this in the user doc..

Reply:
Good idea! (y)